### PR TITLE
Allow to disable scoreboards api

### PR DESCRIPTION
**Pull request by xxDark**

### CODE REVIEW

File: BungeeCord-Patches/0054-Allow-to-disable-scoreboards-api.patch [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
++    }
++    // Waterfall end
++
+     @Data
+     private static class ProtocolData
+     {
+ 
+         private final int protocolVersion;
+         private final TObjectIntMap<Class<? extends DefinedPacket>> packetMap = new TObjectIntHashMap<>( MAX_PACKET_ID );
++        private final PacketData packetData = new PacketData();
```
Comment (MrIvanPlays) [inline BungeeCord-Patches/0054-Allow-to-disable-scoreboards-api.patch:59]: Missing `// Waterfall`

File: BungeeCord-Patches/0054-Allow-to-disable-scoreboards-api.patch [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
++            }
++            Constructor<? extends DefinedPacket> constructor = packetData.packetConstructors[id];
+             try
+             {
+                 return ( constructor == null ) ? null : constructor.newInstance();
+@@ -451,6 +478,7 @@ public enum Protocol
+             try
+             {
+                 Constructor<? extends DefinedPacket> constructor = packetClass.getDeclaredConstructor();
++                constructor.setAccessible( true );
```
Comment (MrIvanPlays) [inline BungeeCord-Patches/0054-Allow-to-disable-scoreboards-api.patch:86]: Missing `// Waterfall`

File: BungeeCord-Patches/0054-Allow-to-disable-scoreboards-api.patch [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+ 
+                 int mappingIndex = 0;
+                 ProtocolMapping mapping = mappings[mappingIndex];
+@@ -477,7 +505,8 @@ public enum Protocol
+ 
+                     ProtocolData data = protocols.get( protocol );
+                     data.packetMap.put( packetClass, mapping.packetID );
+-                    data.packetConstructors[mapping.packetID] = constructor;
++                    PacketData packetData = data.packetData;
```
Comment (MrIvanPlays) [inline BungeeCord-Patches/0054-Allow-to-disable-scoreboards-api.patch:95]: Missing `// Waterfall`

File: BungeeCord-Patches/0054-Allow-to-disable-scoreboards-api.patch [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
++                    packetData.packetConstructors[mapping.packetID] = constructor;
```
Comment (MrIvanPlays) [inline BungeeCord-Patches/0054-Allow-to-disable-scoreboards-api.patch:96]: Missing `// Waterfall`

File: BungeeCord-Patches/0054-Allow-to-disable-scoreboards-api.patch [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+         for ( final ListenerInfo info : config.getListeners() )
+diff --git a/proxy/src/main/java/net/md_5/bungee/command/CommandReload.java b/proxy/src/main/java/net/md_5/bungee/command/CommandReload.java
+index 720d0c3b..74298a77 100644
+--- a/proxy/src/main/java/net/md_5/bungee/command/CommandReload.java
++++ b/proxy/src/main/java/net/md_5/bungee/command/CommandReload.java
+@@ -18,6 +18,7 @@ public class CommandReload extends Command
+     public void execute(CommandSender sender, String[] args)
+     {
+         BungeeCord.getInstance().config.load();
++        BungeeCord.getInstance().reconfigureProtocol();
```
Comment (MrIvanPlays) [inline BungeeCord-Patches/0054-Allow-to-disable-scoreboards-api.patch:206]: Missing `// Waterfall`

File: BungeeCord-Patches/0054-Allow-to-disable-scoreboards-api.patch [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
++import net.md_5.bungee.protocol.packet.ScoreboardScore;
++import net.md_5.bungee.protocol.packet.Team;
+ import net.md_5.bungee.query.RemoteQuery;
+ import net.md_5.bungee.scheduler.BungeeScheduler;
+ import net.md_5.bungee.util.CaseInsensitiveMap;
+@@ -268,6 +272,7 @@ public class BungeeCord extends ProxyServer
+ 
+         pluginManager.loadPlugins();
+         config.load();
++        reconfigureProtocol();
```
Comment (MrIvanPlays) [inline BungeeCord-Patches/0054-Allow-to-disable-scoreboards-api.patch:180]: Missing `// Waterfall`

File: BungeeCord-Patches/0054-Allow-to-disable-scoreboards-api.patch [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+     @Data
+     private static class ProtocolData
+     {
+ 
+         private final int protocolVersion;
+         private final TObjectIntMap<Class<? extends DefinedPacket>> packetMap = new TObjectIntHashMap<>( MAX_PACKET_ID );
++        private final PacketData packetData = new PacketData();
++    }
++
++    private static class PacketData
```
Comment (MrIvanPlays) [inline BungeeCord-Patches/0054-Allow-to-disable-scoreboards-api.patch:62]: The entire PacketData class is missing `// Waterfall` comments

**Review by MrIvanPlays — COMMENTED** [review]
You are missing some things

**Comment by MrIvanPlays** [comment]
And one more thing: how this is useful?

**Comment by mikroskeem** [comment]
>And one more thing: how this is useful?

This should help with high memory usage introduced by storing scoreboard teams in memory.

Personally I find that disabling Scoreboard API is bit invasive, team objects could be improved by for example turning name tag visibility/collision rule flags into enums (which requires de-Lomboking the methods and more care on updates).

**Comment by xism4** [comment]
I tried to disable Scoreboard packets/api and isn't that good, seems that SpigotMC doesn't handle it at all.

**Comment by xxDark** [comment]
> 
> 
> I tried to disable Scoreboard packets/api and isn't that good, seems that SpigotMC doesn't handle it at all.

I'm not maintaining this patch, if you want to achieve the same/better behaivour: https://github.com/xxDark/PassThrough

**Pull request closed without merging**